### PR TITLE
remove conjectured original dates and places

### LINKED ISSUE #19: remove conjectured information from get_original_date()
the whole point of this field is giving the place/date information, as it is written _on the letter_.  
In many cases, though, we have things like `$a[Basel],$c[1743.06.22]`.

Having these editorial information in this field makes the information a bit futile...

@tobiasschweizer We could ignore all fields that start with `[` to solve that. Should I?

### PR DESCRIPTION
**Pull request by BalduinLandolt**
This pull request ignores all conjectured information in `Original date and place`.  
The whole point of this field is to represent date and place __as they are written on the original document__, which makes it really pointless to add conjectured information to this field.

I assume all conjectures to start with square brackets.

Closes #19

### CODE REVIEW

**Comment by BalduinLandolt** [comment]
As far as I can see, this should be ok, like this?

**Comment by tobiasschweizer** [comment]
> The whole point of this field is to represent date and place as they are written on the original document, which makes it really pointless to add conjectured information to this field.

Is there a way to ignore the conjectured information and extract the date as it was originally written? Or would it make more sense to change the data in the source?

**Comment by BalduinLandolt** [comment]
> 
> 
> > The whole point of this field is to represent date and place as they are written on the original document, which makes it really pointless to add conjectured information to this field.
> 
> Is there a way to ignore the conjectured information and extract the date as it was originally written? Or would it make more sense to change the data in the source?

I think, if there is a conjectured date, that means that there is no date in the letter itself at all. (At least the letter ist not explicitely dated in the header; there might be a date in the text somewhere.)  
That's why I chose the approach of saying, if the field starts with `[`, then I treat it as if there was no field at all. (Code lines 448-453.)

And I definitely have no intention of trying to get the others to agree to remove all these fields from BIBB.

I think my approach is actually a neat compromise: We can leave things in the BIBB, so noone complains; but we do not import anything to BEOL that is unnecessary; and actually, it then represents what there really is in the letter: A date, if there is one in the original, and nothing if there is nothing in the original.

File: alephmarcreader/abstractalephmarcreader.py [hunk 1]
```diff
@@ -436,15 +436,22 @@ def get_document_state(self):
...
             date = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'c'), '264', 'c')
             place = self._handle_subfields_cardinality_max_one(self.__get_subfield_texts(field, 'a'), '264', 'a')
-            # TODO Ignore Fields starting wit '['?
+
+            if date and date.startswith('['):
+                date = False
+            if place and place.startswith('['):
+                place = False
+
+            if (date is False) and (place is False):
```
[the remark this answers is not in the document]
Comment (BalduinLandolt) [reply alephmarcreader/abstractalephmarcreader.py]: yes... stylistics. ;)

**Comment by tobiasschweizer** [comment]
Ok, but what would the conjectured date be used for then?

**Comment by BalduinLandolt** [comment]
> 
> 
> Ok, but what would the conjectured date be used for then?

In my opinion: Nothing at all. We have a field for that information... it's called "date". ;)  
But I really don't want to go through the hassle of convincing the others to remove these fields from the BIBB.  
(Not to speak of the "Gruppakorrektuur" that we't have to do, to remove these from BIBB.)

That's why this code fist gets the two relevant fields (lines 445-446) and then checks if the field is conjectured or not (i.e. if it starts with `[` or not). If so, it's treated as if there was nothing there (because that means, there is no original date or place on the actual letter) (lines 448-451). Otherwise, if either place or date are not conjectured, then an object of `OriginalDate` is stored with that information (lines 456-457).

Is that to your liking?

**Pull request merged**